tests/span/test_readable_span.py

```diff
@@ -1,91 +1,109 @@
-    assert span_context.span_id == span_id
-
-    assert span.start_time
-    assert isinstance(span.start_time, int)
-    assert span.start_time > timestamp
-    assert not span.end_time
-    assert not span.attributes
-    assert not span.events
-    assert not span.parent_id
-    assert not span.duration
-    assert span.status
-
-    assert not span.stack
-    assert span.synthetic is False
-
-
-def test_readablespan_with_params(
-    span_context: SpanContext,
-) -> None:
-    span_name = "test-span"
-    parent_id = "123456789"
-    start_time = time.time_ns()
-    end_time = time.time_ns()
-    attributes = {"key": "value"}
-    event_name = "event"
-    events = [Event(event_name, attributes, start_time)]
-    status = Status(StatusCode.OK)
-    stack = ["span-1", "span-2"]
-    span = ReadableSpan(
-        span_name,
-        span_context,
-        parent_id,
-        start_time,
-        end_time,
-        attributes,
-        events,
-        status,
-        stack,
-    )
-
-    assert span.name == span_name
-    assert span.parent_id == parent_id
-    assert span.start_time == start_time
-    assert span.end_time == end_time
-    assert span.attributes == attributes
-    assert span.events == events
-    assert span.status == status
-    assert span.duration == end_time - start_time
-    assert span.stack == stack
+class TestEvent:
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
+        self.test_event = None
+        yield
+
+    def test_event_default(self) -> None:
+        name = "sample-event"
+        self.test_event = Event(name)
+
+        assert self.test_event.name == name
+        assert not self.test_event.attributes
+        assert self.test_event.timestamp < time.time_ns()
+
+    def test_event_with_params(self) -> None:
+        name = "sample-event"
+        attributes = ["attribute"]
+        timestamp = time.time_ns()
+        self.test_event = Event(name, attributes, timestamp)
+
+        assert self.test_event.name == name
+        assert self.test_event.attributes == attributes
+        assert self.test_event.timestamp == timestamp
+
+
+class TestReadableSpan:
+    @pytest.fixture(autouse=True)
+    def _resource(self):
+        self.span = None
+        yield
+        if hasattr(self.span, "events"):
+            self.span.events.clear()
+
+    def test_readablespan(
+        self,
+        span_context: SpanContext,
+        trace_id: int,
+        span_id: int,
+    ) -> None:
+        span_name = "test-span"
+        timestamp = time.time_ns()
+        self.span = ReadableSpan(span_name, span_context)
+
+        assert self.span is not None
+        assert isinstance(self.span, ReadableSpan)
+        assert self.span.name == span_name
+
+        span_context = self.span.context
+        assert isinstance(span_context, SpanContext)
+        assert span_context.trace_id == trace_id
+        assert span_context.span_id == span_id
```

Comment:
Instead of overwriting the `span_context` you are passing as argument, this suggestion would be better.
```suggestion
        assert self.span.context == span_context
        assert isinstance(self.span.context, SpanContext)
        assert self.span.context.trace_id == trace_id
        assert self.span.context.span_id == span_id
```